msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/AbstractMsalUiTest.java

```diff
@@ -0,0 +1,314 @@
+        return new AuthenticationCallback() {
+            @Override
+            public void onSuccess(IAuthenticationResult authenticationResult) {
+                Assert.assertFalse(StringUtil.isEmpty(authenticationResult.getAccessToken()));
+                mAccount = authenticationResult.getAccount();
+                final Map<String, ?> claims = authenticationResult.getAccount().getClaims();
+                Assert.assertTrue(claims.containsKey(requestedClaim));
+                if (!TextUtils.isEmpty(expectedValue)) {
+                    final Object claimValue = claims.get(requestedClaim);
+                    Assert.assertEquals(expectedValue, claimValue.toString());
+                }
+                latch.countDown();
+            }
+
+            @Override
+            public void onError(MsalException exception) {
+                junit.framework.Assert.fail(exception.getMessage());
+                latch.countDown();
+            }
+
+            @Override
+            public void onCancel() {
+                junit.framework.Assert.fail("User cancelled flow");
+                latch.countDown();
+            }
+        };
+    }
+
+    /**
+     * A callback that can be used to perform assertions on completion of an interactive request
+     * (cancel case) test.
+     *
+     * @param latch the latch associated to this request
+     * @return an {@link AuthenticationCallback} object
+     */
+    protected AuthenticationCallback cancelInteractiveCallback(final CountDownLatch latch) {
+        return new AuthenticationCallback() {
+            @Override
+            public void onSuccess(IAuthenticationResult authenticationResult) {
+                Assert.fail("Unexpected Success!");
+                latch.countDown();
+            }
+
+            @Override
+            public void onError(MsalException exception) {
+                Assert.fail(exception.getMessage());
+                latch.countDown();
+            }
+
+            @Override
+            public void onCancel() {
+                latch.countDown();
+            }
+        };
+    }
+
+    /**
+     * A callback that can be used to perform assertions on completion of an silent request
+     * (success case) test.
+     *
+     * @param latch the latch associated to this request
+     * @return an {@link AuthenticationCallback} object
+     */
+    protected SilentAuthenticationCallback successfulSilentCallback(final CountDownLatch latch) {
+        return new SilentAuthenticationCallback() {
+            @Override
+            public void onSuccess(IAuthenticationResult authenticationResult) {
+                Assert.assertFalse(StringUtil.isEmpty(authenticationResult.getAccessToken()));
+                mAccount = authenticationResult.getAccount();
+                latch.countDown();
+            }
+
+            @Override
+            public void onError(MsalException exception) {
+                junit.framework.Assert.fail(exception.getMessage());
```

Comment:
Very minor nit: could be `Assert.fail` (assuming this is imported)

Reply:
Addressed